python-package/xgboost/sklearn.py

```diff
@@ -904,6 +908,17 @@ def _duplicated(parameter: str) -> None:
 
         return model, metric, params, early_stopping_rounds, callbacks
 
+    def _create_dmatrix(self, ref: Optional[DMatrix], **kwargs: Any) -> DMatrix:
+        # Use `QuantileDMatrix` to save memory.
+        if self.tree_method in ("hist", "gpu_hist"):
+            try:
+                return QuantileDMatrix(
+                    **kwargs, ref=ref, nthread=self.n_jobs, max_bin=self.max_bin
+                )
+            except TypeError:  # `QuantileDMatrix` supports lesser types than DMatrix
```

Comment:
Which cases are not supported out of curiosity? Just wondering if this is going to run into the exception regularly in some cases.

Reply:
datatable, CSC, arrow.  Also, the DMatrix has a dispatcher to convert unknown types to scipy csr.